docs/reference/method/MongoDBClient-addSubscriber.txt

```diff
@@ -0,0 +1,130 @@
+
+   .. code-block:: php
+
+      function addSubscriber(MongoDB\Driver\Monitoring\Subscriber $subscriber): void
+
+   This method has the following parameters:
+
+   .. include:: /includes/apiargs/MongoDBClient-method-addSubscriber-param.rst
+
+   .. note::
+
+         If $subscriber is already registered with this Client, this function
+         is a no-op. If $subscriber is also registered globally, it will still
+         only be notified once of each event for this Client.
+
+Errors/Exceptions
+-----------------
+
+.. include:: /includes/extracts/error-invalidargumentexception.rst
+
+Example
+-------
+
+Create a :phpclass:`MongoDB\\Driver\\Monitoring\\CommandSubscriber` that
+logs all events:
+
+.. code-block:: php
+
+    <?php
+
+        use MongoDB\Driver\Monitoring\CommandSubscriber;
+        use MongoDB\Driver\Monitoring\CommandStartedEvent;
+        use MongoDB\Driver\Monitoring\CommandSucceededEvent;
+        use MongoDB\Driver\Monitoring\CommandFailedEvent;
+
+        class LogCommandSubscriber implements CommandSubscriber
+        {
+            private $stream;
+            public function __construct($stream)
+            {
+                $this->stream = $stream;
+            }
+
+            public function commandStarted(CommandStartedEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Started command #%d "%s": %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    Document::fromPHP($event->getCommand())->toCanonicalExtendedJSON(),
+                    PHP_EOL,
+                ));
+            }
+
+            public function commandSucceeded(CommandSucceededEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Succeeded command #%d "%s" in %d microseconds: %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    $event->getDurationMicros(),
+                    json_encode($event->getReply()),
+                    PHP_EOL,
+                ));
+            }
+
+            public function commandFailed(CommandFailedEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Failed command #%d "%s" in %d microseconds: %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    $event->getDurationMicros(),
+                    $event->getError()->getMessage(),
+                    PHP_EOL,
+                ));
+            }
+        }
+
+The subscriber can then be registered with a Client:
+
+.. code-block:: php
+
+    <?php
+
+        $client = new MongoDB\Client();
+        $subscriber = new LogCommandSubscriber(STDERR);
+
+        $client->addSubscriber($subscriber);
+
+        $client->test->users->insertOne(['username' => 'alice']);
+
+The above code will log the following to err output:
```

Comment:
```suggestion
The above code will write the following to stderr output:
```

I'd refrain from using "log" here, since command logging will eventually be implemented down the road. That will be different from APM in that the driver will generate the log message on its own.